Update pull mode doc (#578)
awslabs/soci-snapshotter: docs/pull-modes.md

A layer will be mounted as a fuse mountpoint if it's being lazily loaded, or as
a normal overlay layer if it's not.

Overall, lazily pulling a container image with soci-snapshotter
(via the `soci image rpull` command) involves the following steps:

> ## Step 1: specify SOCI index digest

Reviewer: Nit

```suggestion
## Step 1: Optionally specify a SOCI index digest
```

SOCI index manifest and will be passed to the snapshotter.

If not provided, the snapshotter will use the OCI distribution-spec's
[Referrers API](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers)
(if available, otherwise the spec's
[fallback mechanism](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#unavailable-referrers-api))
to fetch a list of available indices. An index will be chosen from the list of available indices,
but the selection process is undefined and it may not choose the same index every time.

> Check out [this doc](./getting-started.md#lazily-pull-image) for how to
> validate if this step is successful or not, and [the debug doc](./debug.md#common-scenarios)
> for the common scenarios where `rpull` might fail and how to debug/fix them.

## Step 2: fetch SOCI artifacts

During `rpull`, on the first layer mount there will be an attempt to download
and parse the SOCI manifest. If this doesn’t go well, there will be the following
error in the log: `unable to fetch SOCI artifacts:`, indicating that the
container image will not be lazily loaded. In this case, the snapshotter will

> fallback to default snapshotter configured (eg: overlayfs) entirely.

Reviewer: nit: the wording here is a bit awkward.
```suggestion
fallback to default snapshotter (eg: overlayfs) for all layers.
```
